submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/ProjectsApiTest.kt

```diff
@@ -0,0 +1,91 @@
+import ac.uk.ebi.biostd.persistence.repositories.UserDataRepository
+import ac.uk.ebi.biostd.persistence.service.SubmissionRepository
+import ebi.ac.uk.dsl.submission
+import ebi.ac.uk.model.extensions.addAccessTag
+import ebi.ac.uk.model.extensions.title
+import ebi.ac.uk.persistence.PersistenceContext
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import io.github.glytching.junit.extension.folder.TemporaryFolderExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.test.annotation.DirtiesContext
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+@ExtendWith(TemporaryFolderExtension::class)
+internal class ProjectsApiTest(tempFolder: TemporaryFolder) : BaseIntegrationTest(tempFolder) {
+    @Nested
+    @ExtendWith(SpringExtension::class)
+    @Import(value = [TestConfig::class, SubmitterConfig::class, PersistenceConfig::class, TestConfig::class])
+    @SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+    @DirtiesContext
+    inner class SingleSubmissionTest(
+        @Autowired val submissionRepository: SubmissionRepository,
+        @Autowired val tagsDataRepository: TagsDataRepository,
+        @Autowired val accessPermissionRepository: AccessPermissionRepository,
+        @Autowired val userDataRepository: UserDataRepository,
+        @Autowired val persistenceContext: PersistenceContext
+        ) {
+        @LocalServerPort
+        private var serverPort: Int = 0
+
+        private lateinit var webClient: BioWebClient
+
+        private val projectAccNo = "SampleProject1"
+
+        private lateinit var accessTag: AccessTag
+
+        @BeforeAll
+        fun init() {
+            val securityClient = SecurityWebClient.create("http://localhost:$serverPort")
+            securityClient.registerUser(GenericUser.asRegisterRequest())
+            webClient = securityClient.getAuthenticatedClient(GenericUser.username, GenericUser.password)
+            accessTag = AccessTag(name = projectAccNo)
+            tagsDataRepository.save(accessTag)
+            accessPermissionRepository.save( AccessPermission(
+                user = userDataRepository.getOne(1), //TODO: Replace by userDataRepository.findByEmail after merging PR#113
+                accessTag = accessTag,
+                accessType = AccessType.ATTACH) )
+        }
+
+        @Test
+        fun `get projects`() {
+            val submission = submission(projectAccNo) {
+                title = "Sample Project"
+            }
+            submission.accessTags.add(projectAccNo)
```

Comment:
The idea of using the DSL is to avoid setting values to the object. For example, this submission would be something like:
```
val submission = submission(projectAccNo) {
    title = "Sample Project"
    accessTags = mutableListOf(projectAccNo)

    section(SubmissionTypes.Project.value) { }
}
```